torch/csrc/api/include/torch/optim/adam.h

```diff
@@ -32,35 +30,33 @@ class Adam : public Optimizer {
 
   void step() override;
 
-  template <class Archive>
-  void serialize(Archive& ar) {
-#if defined(TORCH_USE_CEREAL)
-    ar(CEREAL_NVP(step_buffers_),
-       CEREAL_NVP(exp_average_buffers_),
-       CEREAL_NVP(exp_average_sq_buffers_),
-       CEREAL_NVP(max_exp_average_sq_buffers_));
-#endif // defined(TORCH_USE_CEREAL)
-  }
+  void save(serialize::Writer& writer) const override;
+  void load(serialize::Reader& reader) override;
 
   AdamOptions options;
 
  private:
-#if defined(TORCH_USE_CEREAL)
-  friend class cereal::access;
-#endif // defined(TORCH_USE_CEREAL)
   Adam() : options(0) {}
 
+  template <typename Self, typename Serializer>
+  static void serialize(Self& self, Serializer& serializer) {
+    optim::detail::serialize(serializer, "step_buffers", self.step_buffers_);
+    serializer(
```

Comment:
you don't have names of parameters available here to use instead of indices, do you?